bugshaker/src/main/java/com/github/stkent/bugshaker/flow/email/FeedbackEmailFlowManager.java

```diff
@@ -30,221 +35,270 @@
+				if (EmailCapabilitiesProvider.canSendEmailsWithAttachments
+					(activityReferenceManager.getValidatedActivity().getPackageManager())) {
+					screenshotProvider.getScreenshotUri(activity)
+						.single()
+						.observeOn(AndroidSchedulers.mainThread())
+						.subscribeOn(AndroidSchedulers.mainThread())
+						.subscribe(new Subscriber<Uri>() {
+							@Override
+							public void onCompleted() {
+								// This method intentionally left blank.
+							}
+
+							@Override
+							public void onError(final Throwable e) {
+								final String errorString = "Screenshot capture failed";
+								toaster.toast(errorString);
+								logger.e(errorString);
+
+								logger.printStackTrace(e);
+
+								sendEmailWithoutScreenshot(context, activity);
+							}
+
+							@Override
+							public void onNext(final Uri uri) {
+								LogcatUtil.saveLogcatToFile(context);
+								sendEmailWithScreenshot(activity, uri, Uri.fromFile(LogcatUtil.getLogFile()));
+							}
+
+
+						});
+				}
+				else {
+					sendEmailWithoutScreenshot(context, activity);
+				}
+			}
+			else {
+				final String warningString = "Window is secured; no screenshot taken";
+				toaster.toast(warningString);
+				logger.d(warningString);
+
+				sendEmailWithoutScreenshot(context, activity);
+			}
+		}
+	};
+
+	private void startActivity(Context applicationContext) {
+		File screenshotFile = ScreenshotUtil.getScreenshotFile(applicationContext);
+		String screenshotFileAbsolutePath = screenshotFile.getAbsolutePath();
+
+		if (screenshotFile.exists()) {
+			Intent goToMainActivityIntent = new Intent(applicationContext, MainActivity.class);
+			goToMainActivityIntent.putExtra("uri", screenshotFileAbsolutePath);
+			goToMainActivityIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+			applicationContext.startActivity(goToMainActivityIntent);
+		}
+		else {
+			throw new RuntimeException();
+		}
+	}
+
+	public void onActivityResumed(@NonNull final Activity activity) {
+		dismissDialog();
+		activityReferenceManager.setActivity(activity);
+	}
+
+	public void onActivityStopped() {
+		dismissDialog();
+	}
+
+	public void startFlowIfNeeded(Context context,
+		final boolean ignoreFlagSecure
+	) {
+		final Logger logger = new Logger(true);
+
+		if (isFeedbackFlowStarted()) {
+			logger.d("Feedback flow already started; ignoring shake.");
+			return;
+		}
+
+		this.ignoreFlagSecure = ignoreFlagSecure;
+
+		showDialog(context);
+	}
+
+	private boolean isFeedbackFlowStarted() {
+		return alertDialog != null && alertDialog.isShowing();
+	}
+
+	private void showDialog(Context applicationContext) {
+
+		final Activity currentActivity = activityReferenceManager.getValidatedActivity();
+		if (currentActivity == null) {
+			return;
+		}
+
+		AlertDialog.Builder bugAlertBuilder = new AlertDialog.Builder(currentActivity);
+		bugAlertBuilder.setMessage(applicationContext.getString(R.string.shake_detected) + '\n' + applicationContext
+			.getString(R.string.report_a_bug));
```

Comment:
\n should not be a part of code. Every string message should come from the string file!